float64 err_ang
float64 err_lin
int8 robot_state
int8 id
int8 n